src/components/FeaturePanel/renderers/FoodHygieneRatingScheme.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import { Typography } from '@material-ui/core';
+import { Restaurant } from '@material-ui/icons';
+import { getEstablishmentRatingValue } from '../../../services/fhrsApi';
+
+interface StarRatingProps {
+  stars: number;
+  maxStars: number;
+}
+
+const StarRating = ({ stars, maxStars }: StarRatingProps) => {
+  const starArray = new Array(maxStars).fill(0);
+  return (
+    <div>
+      {starArray.map((_, i) => (
+        <span>{i < stars ? '★' : '☆'}</span>
+      ))}
+    </div>
+  );
+};
+
+const useLoadingState = () => {
+  const [rating, setRating] = useState<number>();
+  const [error, setError] = useState<string>();
+  const [loading, setLoading] = useState(true);
+
+  const finishRating = (payload) => {
+    setLoading(false);
+    setRating(payload);
+  };
+
+  const startRating = () => {
+    setLoading(true);
+    setRating(undefined);
+    setError(undefined);
+  };
+
+  const failRating = () => {
+    setError('Could not load rating');
+    setLoading(false);
+  };
+
+  return { rating, error, loading, startRating, finishRating, failRating };
+};
+
+export const FoodHygieneRatingSchemeRenderer = ({ v }) => {
+  const { rating, error, loading, startRating, finishRating, failRating } =
+    useLoadingState();
+
+  useEffect(() => {
+    const loadData = async () => {
+      startRating();
+      const ratingValue = await getEstablishmentRatingValue(v);
+      if (Number.isNaN(rating)) {
+        failRating();
+      }
+      finishRating(ratingValue);
+    };
+
+    loadData();
+  }, []);
+
+  return (
+    <>
+      {loading ? (
+        <>
+          <span className="dotloader" />
+          <span className="dotloader" />
+          <span className="dotloader" />
+        </>
+      ) : (
+        <>
+          <Restaurant fontSize="small" />
+          {Number.isNaN(rating) || error ? (
+            <Typography color="error">No rating available</Typography>
+          ) : (
+            <>
+              <StarRating stars={rating} maxStars={5} />
```

Comment:
here: https://osmapp-git-fork-dlurak-foodhygieneratingscheme-zbycz.vercel.app/way/193321876
<img width="563" alt="image" src="https://github.com/zbycz/osmapp/assets/385047/b9b896e0-5135-487e-8cf8-da650a8da3f9">


Could we please make it clear for the user what they are looking at? I am not sure how is this information is usually presented in the UK.. is "FHRS" generally known?

In any case an [InfoButton](https://github.com/zbycz/osmapp/blob/master/src/components/FeaturePanel/EditDialog/SuccessContent.tsx#L44) with a tooltip is a must have. Any foreigner clicking through UK would be confused without it.

Perhaps: `(restaurant icon) FHRS ***** (i)` 
and tooltip like "Food Hygiene Rating System, https://ratings.food.gov.uk/" ?

Reply:
I'm myself not from the uk so I don't know how well it's known.

Reply:
I see, I found that "FHRS" is usually recognized. Let's try the `(restaurant icon) FHRS ***** (i)` structure.

Or maybe just `(restaurant icon) FHRS *****` and title above the row could suffice.

Reply:
Maybe the tooltip should lead the specific establishment, so something like [this](https://ratings.food.gov.uk/business/295203)?

Reply:
Great, i didn't know there is a detail page. In that case i think the text "FHRS *****" could lead to that page. The link can have also a tooltip just with "Food Hygiene Rating System" (translation not needed as it is UK only).